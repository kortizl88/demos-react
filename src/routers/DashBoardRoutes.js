import React, { Fragment } from 'react'
import { Redirect, Route } from "react-router-dom";
import { Switch } from 'react-router-dom'
import { DcScreen } from '../componentes/dc/DcScreen'
import { MarvelScreen } from '../componentes/marvel/MarvelScreen'
import { SearchScreen } from '../componentes/search/SearchScreen';
import { HeroScreen } from '../componentes/ui/heroes/HeroScreen'
import { Navbar } from '../componentes/ui/Navbar'

export const DashBoardRoutes = ({history}) => {
    return (
        <Fragment>
            <Navbar/>
            <div className='container mt-2'>
                <Switch>
                    <Route exact path="/marvel" component={MarvelScreen} />
                    <Route exact path="/hero/:heroId" component={HeroScreen} />
                    <Route exact path="/dc" component={DcScreen} />
                    <Route exact path="/buscar" component={SearchScreen} />

                    <Redirect to="/marvel" />
                </Switch>
            </div>
        </Fragment>
    )
}
