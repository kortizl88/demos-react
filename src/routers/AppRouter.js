import React from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route
} from "react-router-dom";
import { LoginScreen } from "../componentes/login/LoginScreen";
import { Navbar } from "../componentes/ui/Navbar";
import { DashBoardRoutes } from "./DashBoardRoutes";

export const AppRouter = () => {
    return (
        <Router>
      <div>
        <Switch>
          <Route exact path="/login" component={LoginScreen} />
          <Route path="/" component={DashBoardRoutes} />
        </Switch>
      </div>
    </Router>
    )
}
