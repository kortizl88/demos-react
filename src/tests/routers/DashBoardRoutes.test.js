import {mount} from 'enzyme'
import { AuthContext } from '../../auth/AuthContext';
import { DashBoardRoutes } from "../../routers/DashBoardRoutes";
import {MemoryRouter} from 'react-router-dom';

describe('Pruebas en DashBoardRouter', () => {

    const contextValue = {
        dispatch : jest.fn(),
        user: {
            logged: true,
            name: "kris"
        }
    }

    test('Debe mostrarse correctamente', () => {

        const wrapper = mount(
            <AuthContext.Provider value = {contextValue} >
                <MemoryRouter>
                    <DashBoardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();
        expect ( wrapper.find('.text-info').text().trim()).toBe("kris")
    })
    
    
})
