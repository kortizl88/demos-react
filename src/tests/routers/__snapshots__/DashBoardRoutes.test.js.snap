// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en DashBoardRouter Debe mostrarse correctamente 1`] = `ReactWrapper {}`;
